lib/common/idioms.h

```diff
@@ -133,6 +133,10 @@ template<typename A> struct ListItemCount {
 // Given a const reference to a value, return a copy of the value.
 template<typename A> A Clone(const A &x) { return x; }
 
+template<typename A> inline bool PointeeComparison(const A *x, const A *y) {
+  return x == y || (x != nullptr && y != nullptr && *x == *y);
+}
```

Comment:
I kept the name of this function from `lib/evaluate/type.cc` but I would like something shorter, like just `Equals`. Any opinions on that?

Reply:
How about `SameValue` or `EqualValue` ?

Reply:
Please use a name that indicates that pointers/pointees are being compared.  `Equals` is vague.

`ComparePointees` is more what I had in mind.

Reply:
I'd suggest either IsPointerEqual or IsPointeeEqual.  The latter seems more accurate, but it also seems like a typo.

Reply:
The problem is that both pointers and pointees are being compared so either word is slightly misleading. Using "pointer" implies no dereferencing and using "pointee" implies unconditional dereferencing.

Reply:
It appears to be testing if the pointers are referentially transparent, whether those pointers are valid for dereference or not.